.github/scripts/verifyPodfile.sh

```diff
@@ -11,9 +16,97 @@ echo "Podfile: $podfileSha"
 echo "Podfile.lock: $podfileLockSha"
 
 if [ "$podfileSha" == "$podfileLockSha" ]; then
-    echo -e "${GREEN}Podfile verified!${NC}"
-    exit 0
+    echo -e "${GREEN}Podfile checksum verified!${NC}"
 else
-    echo -e "${RED}Error: Podfile.lock out of date with Podfile. Did you forget to run \`npx pod-install\`?${NC}"
+    echo -e "${RED}Error: Podfile.lock checksum mismatch. Did you forget to run \`npx pod-install\`?${NC}"
     exit 1
 fi
+
+# Make sure package.json and package-lock.json are committed to avoid testing issues
+if [ -n "$(git diff --name-only HEAD -- package.json package-lock.json)" ]; then
+  echo -e "${RED}Error: Uncommitted changes to package.json or package-lock.json.  Commit these before continuing!${NC}"
+  exit 1
+fi
+
+# Make sure valid branch ref was passed for removal check
+(git cat-file -e "$1":package-lock.json 2> /dev/null && [ -n "$1" ]) || \
+{ echo -e "${RED}Error: Must specify valid branch name to compare with for Pod removal check.${NC}"; exit 1; }
+
+# If npm packages were not modified in the feature branch we can skip the remaining checks
+if [ -z "$(git diff --name-only .."$1" package-lock.json package.json)" ]; then
+  echo -e "${GREEN}No changes to npm packages were detected.${NC}"
+  exit 0;
+fi
+
+echo -e "${YELLOW}Verifying that pods are synced...${NC}"
+
+# Extracts an array of podspec paths from the config command
+_grab_specs() {
+  npx react-native config | jq '[.dependencies[].platforms.ios.podspecPath | select( . != null )]'
+}
+
+# Maps an array of podspec paths to a raw list of formatted `Pod (version)`s
+_formatted_pod_list() {
+    jq --raw-output --slurp 'map((.name + " (" + .version + ")")) | .[]' <<< "$( \
+      jq -n '$specs | .[]' --argjson specs "$1" | \
+      xargs -L 1 pod ipc spec --silent
+    )"
+}
+
+# Store an array of the feature branch's podspecs
+feature_branch_specs="$(_grab_specs)"
+
+## Verbose output
+[ "$2" == "-v" ] && echo -e "Feature branch specs:\n$feature_branch_specs"
+
+# Grab the podspecs from the main branch by checking out the diffed npm packages
```

Comment:
You lost me here ... why do we need to checkout the main branch at all?

Reply:
I think we should be able to verify that all the correct pods are installed by looking at only the feature branch right? Using `npx react-native config` as the source of truth.

Reply:
To determine removals. The script checks out the package.json from the main branch and runs the `react-native config` command after npm install.  That way we can do an array subtraction between the packages on the feature branch and the main branch to find out if anything was removed.  Since the dependency graph would likely not be changed significantly it's a quick procedure.

Reply:
@roryabraham It's straight forward to determine what pods should be in the Podfile.lock using the config command.  However, it's much harder to determine what should not be in the Podfile.lock by referencing just the config command on the feature branch. It only displays currently installed packages, so we have nothing to reference against to quickly determine what has been removed without looking at the main branch.

Adapting the solution may be possible if we got into parsing the Podfile.lock file and attempting to count packages, or filter what packages weren't displayed by the config command. However I feel like it would introduce more complexity than it's worth.  At that point we'd probably be better off leveraging the dependency resolution of the pod install command and comparing the lockfile that way.

Reply:
Looking at the Podfile.lock, it does display a list of paths to the corresponding node_module directories for each podspec.  We may be able to parse these and determine if the directory exists to determine removals.  I can experiment with this method of determining removals if you would prefer.

Reply:
After some experimenting, it looks like we could use the yq library to parse the Podfile.lock's yaml and determine all of the directories that should contain podspecs.  So we might be able to just do `cat ios/Podfile.lock | yq '.["EXTERNAL SOURCES"].[].":path" | select ( . == "*node_modules*" )'` and then just check for the existence of these directories to determine if something was removed.  I will implement this in the script to see if it still works as expected.

Reply:
This technique appears to work very well.  I was able to simplify the script by leveraging yq to determine any removed packages that were still contained in the Podfile.lock.  It isolates the check to the feature branch which is what I was hoping to accomplish.  The latest revision has been pushed.